src/crimson/osd/watch.cc

```diff
@@ -122,15 +122,18 @@ seastar::future<> Watch::notify_ack(
   const uint64_t notify_id,
   const ceph::bufferlist& reply_bl)
 {
-  logger().info("{}", __func__);
-  return seastar::do_for_each(in_progress_notifies,
-    [this_shared=shared_from_this(), reply_bl] (auto notify) {
-      return notify->complete_watcher(this_shared, reply_bl);
-    }
-  ).then([this] {
-    in_progress_notifies.clear();
-    return seastar::now();
-  });
+  logger().info("{} notify_id={}", __func__, notify_id);
+  const auto it = in_progress_notifies.find(notify_id);
+  assert(it != std::end(in_progress_notifies));
```

Comment:
This resolves the `TestLibRBD.QuiesceWatchTimeout` for me locally, what do you think?
```suggestion
  if (it == std::end(in_progress_notifies)) {
     logger().debug("notify_ack nothing to do");
     return seastar::now();
  }
```

Reply:
Hmm, we shouldn't be getting a notify ack from something not in the in_progress_notifies list, I don't think.  Perhaps make it error() and debug it after this PR merges?

Reply:
Switched to logging an error.